Adapter/GD.php

```diff
@@ -40,6 +40,12 @@ public function __construct()
             throw new \RuntimeException('You need to install GD PHP Extension to use this library');
         }
     }
+    
+    public function __destruct()
+    {
+        if (null !== $this->resource)
+            imagedestroy($this->resource);
```

Comment:
I cannot recommend using `imagedestroy()` as the resource may have been used elsewhere.